File: components/credentials/src/main/java/org/cloudfoundry/credhub/audit/entities/GetCertificateByName.java
package org.cloudfoundry.credhub.audit.entities;

import org.cloudfoundry.credhub.audit.OperationDeviceAction;
import org.cloudfoundry.credhub.audit.RequestDetails;

public class GetCertificateByName implements RequestDetails {
  private String name;

  public String getName() {
    return name;
  }

  public void setName(final String name) {
    this.name = name;
  }

  @Override
  public OperationDeviceAction operation() {
    return OperationDeviceAction.GET_CERTIFICATE;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/audit/entities/GetCertificateByName.kt
package org.cloudfoundry.credhub.audit.entities

import org.cloudfoundry.credhub.audit.OperationDeviceAction
import org.cloudfoundry.credhub.audit.RequestDetails

class GetCertificateByName : RequestDetails {
    var name: String? = null

    override fun operation(): OperationDeviceAction = OperationDeviceAction.GET_CERTIFICATE
}
